### PR TITLE
RooFit::MultiProcess & TestStatistics part 2 redo: RooFitZMQ & MultiProcess

### PR DESCRIPTION
**Pull request by egpbos**
This PR is a do-over of #8385 and #8412 and, as such, again the second part of a split and clean-up of #8294. The most important blocker in those PRs was the inclusion of a patched libzmq in RooFitZMQ itself. [This patch has now been included in libzmq proper](https://github.com/zeromq/libzmq/pull/4255). Another blocking review comment was that libzmq symbols must not be allowed to be exported through our libraries. [This has been solved in theory, and in practice is pending another PR to libzmq](https://github.com/zeromq/libzmq/pull/4266). Having fixed these two blockers, we should now be able to continue.

To recap:

> In this PR, we introduce two packages: RooFitZMQ and RooFit::MultiProcess. It also adds two builtins for ZeroMQ to ease dependency management: libzmq and cppzmq. The builtin for libzmq is especially necessary at this point because it has recently gained a necessary feature that has not been released yet.
> 
> RooFit::MultiProcess is a task-based parallelization framework.
> 
> It uses forked processes for parallelization, as opposed to threads. We chose this approach because A) the existing RooRealMPFE parallelization framework already made use of forks as well, so we had something to build on and B) it was at the time deemed infeasible to check the entire RooFit code for thread-safety. Moreover, we use MultiProcess to parallelize gradients -- i.e. the tasks to be executed in parallel are partial derivatives -- and these are sufficiently large tasks that communication in between tasks is not a big concern in the big fits that we aimed to parallelize.
> 
> The communication between processes is done using ZeroMQ. The ZeroMQ dependency is wrapped in convenience classes contributed by @roelaaij which here are packaged as RooFitZMQ.

Will un-draft the PR once the following is done (based on previous review comments by @guitargeek @hageboeck @amadio @lmoneta and also some other things from myself):
- [x] includes: correct order (matching header, RooFit, ROOT, std) and ROOT includes in quotation marks
- [x] fix ZMQ deprecation warnings
- [x] refactor member names: underscore suffix
- [x] document important things with doxygen
- [x] remove commented out code and TODOs and other junk
- [x] fix copyright headers + author lists (RooFitZMQ: me, Roel; MP: me, Inti, Vince)
- [ ] rebase in 2-3 neat commits that all compile and pass tests
- [x] clang-tidy up
- [x] change libzmq builtin back to master after PR is merged: https://github.com/zeromq/libzmq/pull/4266
- [ ] ~use `enum class` instead of template parameters for minimizer function implementation choice~ -> next PR

_Edit 18 Nov 2021:_ the following list is to keep track of unaddressed (at time of writing) comments made in this thread (because the thread is so long that it is very inconvenient to navigate on GitHub which doesn't load it all at once):
- [x] https://github.com/root-project/root/pull/9078#pullrequestreview-773656413: only need to rebase, but that is already listed above. 
- [x] https://github.com/root-project/root/pull/9078#pullrequestreview-790026907: we have to double check whether the build issues still exist. They should be gone, because we don't build dictionaries anymore.
- [x] https://github.com/root-project/root/pull/9078#discussion_r736998615: Related to the issue above, iiuc, because the include was missing from the dictionary, so this can probably also be marked resolved now.
- [x] https://github.com/root-project/root/pull/9078#pullrequestreview-791797535: change `inc` to `res` in RooFitZMQ and MultiProcess and only include these zmq header exposing include directories to specific targets that need them using `target_include_directories`. This way, we don't transitively expose zmq includes to ROOT users.
    - [x] https://github.com/root-project/root/pull/9078#pullrequestreview-791786326: The above solution also circumvents this issue with `ZMQ_ENABLE_DRAFT` preprocessor defines.
- [x] https://github.com/root-project/root/pull/9078#pullrequestreview-791883192: change copyright/license headers.

Let me know if you find additional items for the todo list.

### CODE REVIEW

**Review by amadio — DISMISSED** [review]
This has improved a lot over the previous version. I think with just a bit of work to ensure the build system parts work in both builtin and non-builtin mode, it can be merged. I will leave review of things other than the build system for others in the team that are more familiar with RooFit. 

BTW, it would also be really nice to have the final git history with fixups and formatting changes already squashed into the appropriate commits if you can do that. Thanks a lot!

**Review by Axel-Naumann — DISMISSED** [review]
Could you
- ideally state `(c) CERN, 2021` or similar wording
- definitely change the license to match the other parts of RooFit

such that future changes in (c) assignment are not even more problematic than they are now? :sweat_smile: That would be *very* much appreciated! Example:

```
/*
 * Project: RooFit
 * Authors:
 *   PB, Patrick Bos, Netherlands eScience Center, p.bos@esciencecenter.nl
 *   IP, Inti Pelupessy, Netherlands eScience Center, i.pelupessy@esciencecenter.nl
 *
 * Copyright (c) 2021, CERN
 *
 * Redistribution and use in source and binary forms,                        *
 * with or without modification, are permitted according to the terms        *
 * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             */
```

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build on windows10/cxx14

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build on ROOT-ubuntu16/nortcxxmod with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build on ROOT-ubuntu2004/soversion with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
@phsft-bot build on ROOT-ubuntu2004/soversion with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

File: builtins/zeromq/libzmq/CMakeLists.txt
Comment (egpbos) [reply builtins/zeromq/libzmq/CMakeLists.txt]: This is outdated, I will mark it resolved, please unresolve if you disagree.

File: builtins/zeromq/libzmq/CMakeLists.txt
Comment (guitargeek) [reply builtins/zeromq/libzmq/CMakeLists.txt:37]: Nope, not an issue anymore. The builtin zeromq works for me now!

File: cmake/modules/SearchInstalledSoftware.cmake
Comment (guitargeek) [inline cmake/modules/SearchInstalledSoftware.cmake:1901]: Have you checked that this works? When I try to build ROOT without the builtin zeromq but with a new zeromq package on my system (it has `zmq_ppoll`), I still the the error:
```
ERRORDetected ZeroMQ version is too old (does not have zmq_ppoll). Aborting, because fail-on-missing was set.
```
But when I try to compile the `CheckSymbolExists.c` file that cmake produces manually:
```C++
#include <zmq.h>

int main(int argc, char** argv) {
   return ((int*)(&zmq_ppoll))[argc];
}
```
It works! I think somehow the `ZMQ_BUILD_DRAFT_API` flag might not me propagated correctly. Does it work for your system?

**Comment by guitargeek** [comment]
Hi @egpbos, thanks a lot for the updates and great that everything now passes with the CI! I can also build ROOT on my system successfully with the new `zeromq` and `cppzmq` builtins that you introduced.

However, I still have problems builting without the builtins, taking zeromq from the system. I installed zeromq from the GitHub master on my system, to make sure I have `zmq_ppoll`. One problem is that the check for the `zmq_ppoll` symbol doesn't seem to work (see inline comment above), and then the configuration fails with the following error:
```
CMake Error at cmake/modules/SearchInstalledSoftware.cmake:17 (_find_package):
  By not providing "Findcppzmq.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "cppzmq", but
  CMake did not find one.

  Could not find a package configuration file provided by "cppzmq" with any
  of the following names:

    cppzmqConfig.cmake
    cppzmq-config.cmake

  Add the installation prefix of "cppzmq" to CMAKE_PREFIX_PATH or set
  "cppzmq_DIR" to a directory containing one of the above files.  If "cppzmq"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake/modules/SearchInstalledSoftware.cmake:1927 (find_package)
  CMakeLists.txt:245 (include)


-- Configuring incomplete, errors occurred!
```
Is it really reasonable to expect a `cppzmqConfig.cmake` on the system, just for this `zmq.hxx` header that is most likely in the same directory as `zmq.h` anyway?

Maybe there are some specific subtleties on my system that cause the build configuration not to work. Let's see with the CI when we disable builtins but enable roofit_multiprocess. What I would expect is that the configuration realized that an appropriate zeromq version is missing, and it switches to the builtins automatically right?

**Comment by guitargeek** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON

**Comment by egpbos** [comment]
@guitargeek That would/should work if CI didn't build with fail-on-missing. Let's try turning that off then (I realize this completely changes the build setup, hopefully it doesn't break other things, but just to try out what @guitargeek is suggesting).

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

File: cmake/modules/SearchInstalledSoftware.cmake
Comment (egpbos) [reply cmake/modules/SearchInstalledSoftware.cmake]: I've pushed a fix for this, could you check if it works for you now?

**Comment by egpbos** [comment]
> Is it really reasonable to expect a cppzmqConfig.cmake on the system, just for this zmq.hxx header that is most likely in the same directory as zmq.h anyway?

@guitargeek I have added a Findcppzmq module that should find the header. Could you check if it works for you?

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

**Comment by guitargeek** [comment]
> > Is it really reasonable to expect a cppzmqConfig.cmake on the system, just for this zmq.hxx header that is most likely in the same directory as zmq.h anyway?
> 
> @guitargeek I have added a Findcppzmq module that should find the header. Could you check if it works for you?

Thanks for the update! Yes, I confirm that things work for me now in all scenarios:
* switching builtins ON
* switching builtins OFF and having new zeromq with ppoll on the system
* switching builtins OFF, fail on missing OFF and having old zeromq on the system (it automatically switches builtins on then)

**Comment by egpbos** [comment]
I have implemented all requested changes by reviewers @amadio, @Axel-Naumann and @guitargeek. Additionally, bugs have been fixed, build system works and CI passes. I rebased the whole thing into three commits corresponding to the actual three additions: ZeroMQ dependency handling (including builtin), `RooFitZMQ` and `RooFit::MultiProcess`.

I think this ready to merge, unless you have some final review suggestions.

File: roofit/multiprocess/res/RooFit/MultiProcess/Job.h
Comment (lmoneta) [inline roofit/multiprocess/res/RooFit/MultiProcess/Job.h:1]: Why the header files are in the subdirectory multiprocess/res and not in multiprocess/inc ?

**Review by lmoneta — COMMENTED** [review]
The PR looks good. I think we are almost ready to merge it. 
Can the empty Linkdef files be removed and the subdirectory with the headers renamed from res to inc ? 

I see also that one test seems to hang on my machine when trying the PR. 
The test is `test-RooFit-MultiProcess-Messenger`. Can this be checked  ?

**Comment by lmoneta** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dfail-on-missing=OFF

File: roofit/multiprocess/res/RooFit/MultiProcess/Job.h
Comment (guitargeek) [reply roofit/multiprocess/res/RooFit/MultiProcess/Job.h:1]: That was requested by @Axel-Naumann in https://github.com/root-project/root/pull/9078#discussion_r738341165

**Comment by Axel-Naumann** [comment]
@lmoneta 

> renamed from res to inc

We use `res/` for headers that are not meant to be used by users, but are needed by multiple ROOT "modules". `src/` for headers that are only used within that module's sources. And `inc/` for headers that we expect will be included by users. IIUC some of these headers are indeed in the `res/` category?

**Comment by egpbos** [comment]
@lmoneta I have checked `test-RooFit-MultiProcess-Messenger` on infinite repeat on multiple machines and also on CI, it should be bug free now, so I need a bit more information. What setup are you running on? What is the output of the test before it hangs? Did you check out the branch clean (because I rebased / force pushed everything yesterday, so just a pull may not have fast-forwarded the branch)?

I will remove the Linkdef files.

Indeed, the `res` directories are to prevent users from including the MP headers. I may need to do some magic in the next PR where they will need to be included in user facing headers again, but we'll cross that bridge when we get there.

**Comment by egpbos** [comment]
@lmoneta I removed the Linkdef files. Do you have more information on the Messenger test hanging?

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by lmoneta** [comment]
I see the test passes in the CI, so it could be a problem with my environment. 
The hanging test is `TestMPMessenger.StressPubSub`
Here are the last lines of the log:
```[ RUN      ] TestMPMessenger.SigStop
first send message to queue...
PID 493293 sends M2Q 1
sleep for 2 seconds...
PID 493330 receives M2Q 1
received 1 on queue
SIGSTOPping all children...
send another message to queue...
PID 493293 sends M2Q 2
sleep for 2 seconds...
SIGCONT queue and worker 1...
sleep for 2 seconds...
PID 493330 receives M2Q 2
received 2 on queue
PID 493330 sends Q2M 3
SIGCONT worker 2...
PID 493293 receives Q2M 3
Messenger dtor on PID 493293
handled Terminated on PID 493330
handled Terminated on PID 493328
handled Terminated on PID 493329
Messenger dtor on PID 493330
Messenger dtor on PID 493328
Messenger dtor on PID 493329
[       OK ] TestMPMessenger.SigStop (6004 ms)
[ RUN      ] TestMPMessenger.StressPubSub
Messenger dtor on PID 493525
```

**Comment by lmoneta** [comment]
However I must say that sometimes (after 30 sec 1 minutes) in some case the test ends, I see however lots of memory used and lots of processes spawned. Can you make a lighter and faster test ?

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

**Comment by egpbos** [comment]
After considering with @lmoneta, I disabled the hanging stress test. It was meant to test the pub-sub connection between master and workers, but this is in fact already done in the `Messenger` constructor now. The ctor connection check is also more robust than the stress test, so it is unnecessary.

I also removed remaining debug prints.

Ready to merge, from my side.

**Comment by egpbos** [comment]
@phsft-bot build with flags -Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON

File: builtins/zeromq/cppzmq/CMakeLists.txt
Comment (guitargeek) [inline builtins/zeromq/cppzmq/CMakeLists.txt:26]:
```suggestion
```
The `zmq.hpp` header should not be installed, as it is only used for building ROOT (just like the new roofit libraries introduced in this PR).

File: .ci/copy_headers.sh
Comment (guitargeek) [reply .ci/copy_headers.sh:13]: We can revert this change now, because `zmq.hpp` is not included in any RooFit interfaces.

**Review by guitargeek — APPROVED** [review]
Thank you very much @egpbos for your contribution and addressing the review comments!

* the new roofit/multiprocess` and `roofit/roofitZMQ` libraries are introduced. They don't install any headers because they will only be introduced as build dependencies of RooFit in the future
* the two new builtin dependecies `zeromq` and `cppzmq` are introduced

This is ready to be merged now. I still have two requests about the header installation, but I will address them myself in a follow-up PR.

**Comment by amadio** [comment]
@egpbos Given my [comment](https://github.com/root-project/root/pull/8385#issuecomment-870570029) in your previous merge request, I am quite disappointed to learn that this has been merged as enabled by default when it depends on a yet unreleased version of ZeroMQ. This, for example, has caused failures in pretty much all the LCG builds today: https://cdash.cern.ch/index.php?project=LCGSoft&date=2021-11-29. Please disable `roofit_multiprocess` by default and re-enable it only when a *released version* of ZeroMQ has the functionality you need, adding the proper version in your call to `find_package()`.

**Comment by egpbos** [comment]
@amadio You're right, we overlooked that.

We should discuss in the RooFit dev meeting how to do this, also in the long run. I guess for the time being, setting it to `OFF` by default sounds reasonable. What do you think @guitargeek @lmoneta?

If this is indeed what we do, it would be quite important (imho) that CI still builds with `-Droofit_multiprocess=ON -Dbuiltin_zeromq=ON -Dbuiltin_cppzmq=ON` so that future changes don't break it. Then, when the new `libzmq` version is released, it will be easy to adapt the CMake config to that.

**Comment by guitargeek** [comment]
Hi @amadio! I thought we had addressed this problem by making the unreleased `zeromq` interface only a dependency at ROOT build time, and the libraries are linked statically against this unreleased `zeromq` library. So it should not matter what is installed on the system, as long as the builtins are used for building.

What exactly is the problem you have at configuration times in these LCG builds? I can't see the full log. I hope it's just one of your configurations checks that doesn't work in your environment and we can fix that.

**Comment by egpbos** [comment]
Ah, I wasn't aware that that was also meant to solve @amadio's problem, thx @guitargeek.

**Comment by amadio** [comment]
The problem is that LCG cannot use builtins (and also sets `fail-on-missing=ON`), as well as other Linux distributions, like Fedora, Gentoo, Arch, etc. Something that only works with builtins cannot be enabled by default.

**Comment by guitargeek** [comment]
Makes sense in general, that building without builtins should be possibe in LCG and distros! But why can't we use builtins at compile time only? I don't see why not, and that would also be a big restriction :)

**Comment by Axel-Naumann** [comment]
> The problem is that LCG cannot use builtins

That's only half-true, of course we do have builtins that LG builds with just fine (libAfterImage, llvm, etc). But then we must not expose it as "override-able". So either it's not configurable and we always build our "hacked" zmq (exporting neither headers nor symbols, so the world doesn't care), or we wait until it's released.

**Comment by guitargeek** [comment]
100 % agreed. So I'll open a PR to always use our "hacked" zmq. When it's release we can change it back

**Comment by amadio** [comment]
> But why can't we use builtins at compile time only?

ZeroMQ is already part of LCG, so unless we are *absolutely sure* that no symbol will be leaked, we should use the external one. Like with LLVM, which has bitten with this many times because of mesa using it too and linking to it, there's always the risk of having multiple incompatible versions of symbols loaded into the same running process. Builtins are evil, and patched builtins are extra evil. Please just disable it by default, else you are running the risk of breaking experiment software that uses both ROOT and ZeroMQ because of enabling the builtin. Remember what builtin LZMA without optimizations has caused to CMS already and please don't repeat that mistake.

**Comment by amadio** [comment]
As for requiring a patched version of something, have a look a this bug: https://bugs.gentoo.org/824018 for an example of the consequences.

**Comment by Axel-Naumann** [comment]
What's the release schedule for zeromq versus our mid January ROOT v6.26?

**Comment by guitargeek** [comment]
Ok fair enough.

* I was just checking, and indeed there are `zmq` symbols that end up in `libRooFitMultiprocessing. and `libRooFitZMQ`. There are probably some visibilty-related flags that would need to be changed.

* This means that for the time being, we have to disable `roofit_multiprocessing` by default, even if we always use our "hacked" zeromq at compile time like I suggested before.

* If `roofit_multiprocessing` were to be re-enabled by default, the symbol leak needs to be fixed - including unit tests that make sure there are no `zmq` symbols exported by the new RooFit libraries.

Is that okay?

**Comment by egpbos** [comment]
Sounds reasonable to me. I checked previously that there were no ZeroMQ symbols in RooFitZMQ, so I'm surprised they are back. Indeed, adding a unit test seems ideal to make sure they stay out.

On CI, we can still enable `roofit_multiprocessing`, though, right? That way, at least the code won't rot until we can fix it.

**Comment by egpbos** [comment]
> What's the release schedule for zeromq versus our mid January ROOT v6.26?

@Axel-Naumann I don't know actually, I'll ask on Gitter.

**Comment by egpbos** [comment]
@guitargeek How did you find zmq symbols in the libraries? I don't see them with the following check:

```
root [0] R__LOAD_LIBRARY(libRooFitZMQ)
root [1] int a, b, c; zmq_version(&a, &b, &c);

RooFit v3.60 -- Developed by Wouter Verkerke and David Kirkby
                Copyright (C) 2000-2013 NIKHEF, University of California & Stanford University
                All rights reserved, please read http://roofit.sourceforge.net/license.txt

IncrementalExecutor::executeFunction: symbol 'zmq_version' unresolved while linking [cling interface function]!
root [2]
```

**Comment by guitargeek** [comment]
I have seen them with this command (I got it from [here](https://metricpanda.com/rival-fortress-update-22-finding-duplicate-static-symbols-in-shared-libraries/)):
```bash
nm libRooFitZMQ.so  | c++filt | cut -d ' ' -f 3-99 | sort | uniq
```

I think this was a false alarm, because I forgot to look only for dynamic symbols (passing the `-D` flag to `nm`). I need to read a little bit about linkers to understand what the difference really is :)

But indeed, I think there is no symbol leak because I can't use `libRooFitZMQ` instead of `libzmq` when compiling this code snippet:
```C
#include <zmq.h>

int main(int argc, char** argv) {
   return ((int*)(&zmq_poll))[argc];
}
```
Compiled with `g++ -L.RooFit.so -o check check.C`.

Meaning that as far as I can tell, there is no symbol leak. Still, we have switched off `roofit_multiprocess` by default now. Probably we will only consider enabling it in the release after 6.26.

**Comment by amadio** [comment]
> I need to read a little bit about linkers to understand what the difference really is :)

Just read `man nm`. You have to ensure that all symbols from ZeroMQ are marked with a small `t`, as that means that the symbol is local (i.e. not exported). You can do a crude check for exported symbols like this:
```sh
$ nm libRooFitZMQ.so | awk '/ T /{print $3}' | c++filt
```
and look for ZeroMQ symbols in the output.

**Comment by egpbos** [comment]
Indeed, this shows no ZeroMQ symbols in `libRooFitZMQ.so` or `libRooFitMultiProcess.so`, nor in the new `libRooFitCore.so` of #9349, so it seems we are safe.

@guitargeek Indeed, in #9349, `roofit_multiprocess` will be switched off. In the coming release, people can start trying it out and it could be defaulted ON later on, when the ZeroMQ release with `zmq_ppoll` has also happened.

**Pull request merged**
